File: modules/web-console/backend/test/unit/Utils.test.js
/*
 * Copyright 2019 GridGain Systems, Inc. and Contributors.
 *
 * Licensed under the GridGain Community Edition License (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const assert = require('chai').assert;
const injector = require('../injector');

let utils;
let errors;
let db;

suite('UtilsTestsSuite', () => {
    suiteSetup(() => {
        return Promise.all([injector('services/utils'),
            injector('errors'),
            injector('dbHelper')])
            .then(([_utils, _errors, _db]) => {
                utils = _utils;
                errors = _errors;
                db = _db;
            });
    });

    setup(() => db.init());

    test('Check token generator', () => {
        const tokenLength = 16;
        const token1 = utils.randomString(tokenLength);
        const token2 = utils.randomString(tokenLength);

        assert.equal(token1.length, tokenLength);
        assert.equal(token2.length, tokenLength);
        assert.notEqual(token1, token2);
    });
});

File: modules/configuration-annotation-processor/src/test/java/org/apache/ignite/internal/configuration/processor/UtilsTest.java
package org.apache.ignite.internal.configuration.processor;/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import org.apache.ignite.configuration.annotation.Config;
import org.apache.ignite.configuration.annotation.ConfigurationRoot;
import org.apache.ignite.internal.configuration.processor.Utils;
import org.junit.jupiter.api.Test;

import static org.apache.ignite.internal.configuration.processor.Utils.joinSimpleName;
import static org.apache.ignite.internal.configuration.processor.Utils.simpleName;
import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * Class for testing the {@link Utils}.
 */
public class UtilsTest {
    /** */
    @Test
    void testSimpleName() {
        assertEquals("@Config", simpleName(Config.class));
    }

    /** */
    @Test
    void testJoinSimpleName() {
        assertEquals("@Config", joinSimpleName(" and ", Config.class));
        assertEquals("@Config or @ConfigurationRoot", joinSimpleName(" or ", Config.class, ConfigurationRoot.class));
        assertEquals("", joinSimpleName(" or "));
    }
}
